src/libstd/net/udp.rs

```diff
@@ -450,23 +450,75 @@ impl UdpSocket {
         self.0.ttl()
     }
 
+    /// Specify an IPv4 multicast group for this socket to join.
+    ///
     /// Executes an operation of the `IP_ADD_MEMBERSHIP` type.
     ///
     /// This function specifies a new multicast group for this socket to join.
     /// The address must be a valid multicast address, and `interface` is the
     /// address of the local interface with which the system should join the
     /// multicast group. If it's equal to `INADDR_ANY` then an appropriate
     /// interface is chosen by the system.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::str;
+    /// use std::net::{UdpSocket, Ipv4Addr};
+    ///
+    /// let multiaddr = Ipv4Addr::new(239, 0, 0, 1);
+    /// let inaddr_any = Ipv4Addr::new(0, 0, 0, 0);
+    ///
+    /// let broadcaster = UdpSocket::bind("0.0.0.0:8000").expect("couldn't bind to address");
+    /// let sock = UdpSocket::bind("0.0.0.0:12345").expect("couldn't bind to address");
+    ///
+    /// sock.join_multicast_v4(&multiaddr, &inaddr_any).expect("join_multicast_v4 failed");
+    ///
+    /// broadcaster.send_to(b"world", (multiaddr, 12345)).expect("send_to failed");
+    ///
+    /// let mut buf = [0; 5];
+    /// sock.recv_from(&mut buf).expect("recv_from failed");
+    /// println!("hello {}", &str::from_utf8(&buf).unwrap());
+    /// ```
     #[stable(feature = "net2_mutators", since = "1.9.0")]
     pub fn join_multicast_v4(&self, multiaddr: &Ipv4Addr, interface: &Ipv4Addr) -> io::Result<()> {
         self.0.join_multicast_v4(multiaddr, interface)
     }
 
+    /// Specify an IPv6 multicast group for this socket to join.
+    ///
     /// Executes an operation of the `IPV6_ADD_MEMBERSHIP` type.
     ///
     /// This function specifies a new multicast group for this socket to join.
     /// The address must be a valid multicast address, and `interface` is the
     /// index of the interface to join/leave (or 0 to indicate any interface).
+    ///
+    /// # Note
+    ///
+    /// The `interface` argument is not an `Ipv6Addr` but a u32. This implies
```

Comment:
AFAIK this just corresponds to `ipv6mr_interface`, the underlying field of `ipv6_mreq` corresponding to `IPPROTO_IPV6` and `IPV6_ADD_MEMBERSHIP`. Perhaps we could be sure to mention that here as well?

@APTy your [updated docs](https://github.com/rust-lang/rust/pull/39007/files#r99456228) also look good to me!